index.bs

```diff
@@ -267,43 +267,44 @@ for specific guidance on naming.
 
 <h3 id="feature-detect">New features should be detectable</h3>
 
-The existence of new features should generally be detectable,
-so that web content can act appropriately whether the feature is present or not.
-This applies both to features that
-are not present because they are not implemented,
-and to features that may not be present in a particular configuration
-(ranging from features that are present only on particular platforms
-to features that are
-only available in secure contexts).
-
-It should generally be indistinguishable why a feature is unavailable,
-so that feature detection code written for one case of unavailability
-(e.g., the feature not being implemented yet in some browsers)
-also works in other cases
-(e.g., a library being used in a non-secure context when
-the feature is limited to secure contexts).
-
-However, some features may be for
-interaction with devices that may or may not be available.
-Detection of device availability should generally be separate from feature detection.
-This helps developers who may want to handle device unavailability
-differently from lack of support for the feature.
-It also protects users; for more detail see [[#device-enumeration]].
-
-Detection should always be possible from script,
-but in some cases the feature should also be detectable
-in the language where it is used
+If appropriate, provide a way for authors to programmatically detect
+whether your feature is available,
+so that web content may gracefully handle the feature not being present.
+
+An existing feature may not be available on a page for a number of reasons, including
+- because it has not been implemented yet, or
+- because it's only available in <a href="#secure-context">secure contexts</a>, or
+- because the user denied the page permission to use the feature.
+
+Authors should need to write different code to handle each scenario,
+and should not be able to deduce which cases applied.
+
```

Comment:
I think it would be good to add back the justification, e.g., by adding a second sentence to the paragraph like "This means that an author who writes code to handle one of these scenarios will likely have handled all of them."

Reply:
I think it points out the value to users that, while the author might not care about all of the scenarios, the user gets the value across all of them of the author caring about just one of them.

Reply:
I added an extra paragraph on this.